pkg/controller/worker/machines.go

```diff
@@ -232,3 +234,15 @@ func (w *workerDelegate) generateWorkerPoolHash(pool extensionsv1alpha1.WorkerPo
 	// Generate the worker pool hash.
 	return worker.WorkerPoolHash(pool, w.cluster, additionalHashData...)
 }
+
+// NormalizeLabelsForMachineClass because metadata in OpenStack resources do not allow for certain characters that present in k8s labels e.g. "/",
+// normalize the label by replacing illegal characters with "-"
+func NormalizeLabelsForMachineClass(in map[string]string) map[string]string {
+	notAllowedChars := regexp.MustCompile(`[^a-zA-Z0-9-_:. ]`)
+	res := make(map[string]string)
+	for k, v := range in {
+		newKey := notAllowedChars.ReplaceAllLiteralString(k, "-")
+		res[newKey] = v
```

Comment:
Just the one question: Do we also need to normalize the values?

Reply:
I didn't see the need for it. The values need to follow k8s conventions (enforced by gardener validation) since they are used for node labeling too and I didn't see any issue with that character set when I tried to use it as values, so I would say no.